functions/Upload.js

```diff
@@ -0,0 +1,155 @@
+  var sheet = getCurrentSheet();
+
+  if ( !findColumnByHeaderValue(sheet, HEADER_COMMENTED_PROP_UPLOAD_RELPATH) ) {
+    alertBox(
+      `Add a column ${HEADER_COMMENTED_PROP_UPLOAD_RELPATH} to the header row and try again.`
+    );
+    return
+  }
+  if ( !findColumnByHeaderValue(sheet, HEADER_COMMENTED_PROP_UPLOAD_STATUS) ) {
+    alertBox(
+      `Add a column ${HEADER_COMMENTED_PROP_UPLOAD_STATUS} to the header row and try again.`
+    );
+    return
+  }
+  if ( !findColumnByHeaderValue(sheet, HEADER_COMMENTED_PROP_UPLOAD_CMD) ) {
+    alertBox(
+      `Add a column ${HEADER_COMMENTED_PROP_UPLOAD_CMD} to the header row and try again.`
+    );
+    return
+  }
+
+  var html = HtmlService.createTemplateFromFile("UploaderSideBar.html");
+  var htmlOutput = html
+    .evaluate();
+
+  SpreadsheetApp.getUi().showSidebar(htmlOutput);
+}
+
+function getUploadCredentialsFromFileId(fileId) {
+  var endPoint = getEndpointWrite();
+
+  var url = `${endPoint}/${fileId}@@upload?format=json&frame=object`;
+  var response = restGet(url);
+  var error = response.getResponseCode();
+
+  if (error === 200) {
+    var responseJson = JSON.parse(response.getContentText());
+    return responseJson["@graph"][0]["upload_credentials"];
+
+  } else {
+    Logger.log(`HTTP error ${error}: Failed to get upload credentials from endpoint ${endPoint} for file ID ${fileId}`);
+  }
+}
+
+function getUploadCredentialsFromIdentifyingVal(identifyingVal) {
+  var fileId = `files/${identifyingVal}/`;
+  return getUploadCredentialsFromFileId(fileId);
+}
+
+function initUpload() {
+  if (!checkProfile()) {
+    return;
+  }
+
+  var sheet = getCurrentSheet();
+  var profile = getProfile(getProfileName(), getEndpointRead());
+
+  // check if identifying property exists
+  if (profile["identifyingProperties"]
+    .filter(prop => findColumnByHeaderValue(sheet, prop))
+    .length === 0) {
+    alertBox(
+      `Couldn't find an identifying property (${profile["identifyingProperties"].join(",")}) in header row ${HEADER_ROW}\n\n` +
+      `Add a proper identifying property to the header row and define it for each data row to retrieve from the portal.`
+    );
+    return;
+  }
+
+  const numData = getNumMetadataInSheet(sheet);
+  var results = [];
+
+  for (var row = HEADER_ROW + 1; row <= numData + HEADER_ROW; row++) {
+    var json = rowToJson(
+      sheet, row, keepCommentedProps=true, bypassGoogleAutoParsing=true,
+    );
+
+    // if row is hidden, then skip
+    if (isRowHidden(sheet, row)) {
+      continue;
+    }
+    // if has #skip and it is 1 then skip
+    if (json.hasOwnProperty(HEADER_COMMENTED_PROP_SKIP)) {
+      if (toBoolean(json[HEADER_COMMENTED_PROP_SKIP])) {
+        continue;
+      }
+    }
+    var [identifyingProp, identifyingVal, identifyingCol] =
+      findIdentifyingPropValColInRow(sheet, row, profile);
+
+    if (!identifyingProp || !identifyingVal) {
+      json[HEADER_COMMENTED_PROP_UPLOAD_STATUS] = "Missing identifying val.";
```

Comment:
can you make this more user friendly? Not sure what `"Missing identifying val."` or `val` means if I didn't read the code.

Reply:
fixed